devauth/devauth.go

```diff
@@ -257,6 +266,54 @@ func (d *DevAuth) SubmitAuthRequest(ctx context.Context, r *model.AuthReq) (stri
 
 }
 
+func (d *DevAuth) processPreAuthRequest(ctx context.Context, r *model.AuthReq) (*model.AuthSet, error) {
+	// authset exists?
+	aset, err := d.db.GetAuthSetByDataKey(ctx, r.IdData, r.PubKey)
+	switch err {
+	case nil:
+		break
+	case store.ErrDevNotFound:
+		return nil, nil
+	default:
+		return nil, errors.Wrap(err, "failed to fetch auth set")
+	}
+
+	// if authset status is not 'preauthorized', nothing to do
+	if aset.Status != model.DevStatusPreauth {
+		return nil, nil
+	}
+
+	// auth set is ok for auto-accepting, check device limit
+	allow, err := d.canAcceptDevice(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if !allow {
+		return nil, ErrMaxDeviceCountReached
+	}
+
+	// propagate 'accepted' status to deviceadm
+	sreq := deviceadm.UpdateStatusReq{
+		Status: model.DevStatusAccepted,
+	}
+
+	if err := d.cDevAdm.UpdateStatusInternal(ctx, aset.Id, sreq, d.clientGetter()); err != nil {
+		return nil, errors.Wrap(err, "devadm update status error")
+	}
+
+	// propagate device to inventory
+	if err := d.SubmitInventoryDevice(ctx, model.Device{
+		Id: aset.DeviceId,
+	}); err != nil {
+		return nil, errors.Wrap(err, "inventory device add error")
+	}
+
+	// return the accepted auth set to be persisted
+	aset.Status = model.DevStatusAccepted
+	return aset, nil
+}
```

Comment:
auth set status and device status are not updated

Reply:
fixed